src/objects.rs

```diff
@@ -1048,6 +1090,110 @@ impl GetObjectType for StopPoint {
     }
 }
 
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq)]
+pub struct StopLocation {
+    pub id: String,
+    pub name: String,
+    #[serde(skip)]
+    pub comment_links: CommentLinksT,
+    pub visible: bool,
+    pub coord: Coord,
+    pub parent_id: Option<String>,
+    pub timezone: Option<String>,
+    pub geometry_id: Option<String>,
+    pub equipment_id: Option<String>,
+    pub level_id: Option<String>,
+    #[serde(skip)]
+    pub stop_type: StopType,
+}
+impl_id!(StopLocation);
+impl_comment_links!(StopLocation);
+
+impl AddPrefix for StopLocation {
+    fn add_prefix(&mut self, prefix: &str) {
+        self.id = prefix.to_string() + &self.id;
+        self.parent_id = self.parent_id.as_ref().map(|id| prefix.to_string() + &id);
+        self.geometry_id = self.geometry_id.as_ref().map(|id| prefix.to_string() + &id);
+        self.equipment_id = self
+            .equipment_id
+            .as_ref()
+            .map(|id| prefix.to_string() + &id);
+        self.level_id = self.level_id.as_ref().map(|id| prefix.to_string() + &id);
+    }
+}
+
+#[derivative(Default)]
+#[derive(Derivative, Serialize, Deserialize, Debug, Clone, PartialEq, Eq, Hash)]
+pub enum PathwayMode {
+    #[derivative(Default)]
+    #[serde(rename = "1")]
+    Walkway,
+    #[serde(rename = "2")]
+    Stairs,
+    #[serde(rename = "3")]
+    MovingSidewalk,
+    #[serde(rename = "4")]
+    Escalator,
+    #[serde(rename = "5")]
+    Elevator,
+    #[serde(rename = "6")]
+    FareGate,
+    #[serde(rename = "7")]
+    ExitGate,
+}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Default, Clone)]
+pub struct Pathway {
+    #[serde(rename = "pathway_id")]
+    pub id: String,
+    pub from_stop_id: String,
+    #[serde(skip)]
+    pub from_stop_type: StopType,
```

Comment:
Why do we need this value in the model. This value is already stored in the model if you navigate to the corresponding `StopPoint` with the `from_stop_id`.

Reply:
Location at which the pathway begins. It contains a stop_id that identifies a platform, entrance/exit, generic node or boarding area from the stops.txt file.
I need the location type to sanitize the model.

Reply:
Yes, I understand that but you can look into `collections.stop_points` to get the stop and get the information. But it's always a balance between memory and performance: your proposition is using more memory, mine would need more processing.